crates/autopilot/src/solvable_orders.rs

```diff
@@ -402,18 +427,22 @@ fn get_native_prices(
         .collect()
 }
 
-/// Filters unsigned PreSign and EIP-1271 orders whose signatures are no longer
+/// Finds unsigned PreSign and EIP-1271 orders whose signatures are no longer
 /// validating.
-async fn filter_invalid_signature_orders(
-    mut orders: Vec<Order>,
+async fn find_invalid_signature_orders(
+    orders: &[Order],
     signature_validator: &dyn SignatureValidating,
-) -> Vec<Order> {
-    orders.retain(|order| {
-        !matches!(
+) -> Vec<OrderUid> {
+    let (invalid_orders, orders): (Vec<_>, Vec<_>) = orders.iter().partition(|order| {
```

Comment:
You can use `.partition_map()` instead to avoid re-collecting the `invalid_orders` vector a few lines down.